spec/Section 3 -- Type System.md

```diff
@@ -635,14 +646,14 @@ FieldDefinition : Description? Name ArgumentsDefinition? : Type
 Directives[Const]?
 
 GraphQL operations are hierarchical and composed, describing a tree of
-information. While Scalar types describe the leaf values of these hierarchical
+information. While _leaf types_ describe the leaf values of these hierarchical
 operations, Objects describe the intermediate levels.
 
-GraphQL Objects represent a list of named fields, each of which yield a value of
-a specific type. Object values should be serialized as ordered maps, where the
-selected field names (or aliases) are the keys and the result of evaluating the
-field is the value, ordered by the order in which they appear in the selection
-set.
+GraphQL Objects are a _composite type_ representing a list of named fields, each
+of which yield a value of a specific type. Object values should be serialized as
+ordered maps, where the selected field names (or aliases) are the keys and the
+result of evaluating the field is the value, ordered by the order in which they
+appear in the selection set.
```

Comment:
we might mention here that field names might repeat in this output map, as a result of field merging